psqlgraph/node.py

```diff
@@ -142,6 +142,26 @@ def copy(self):
         )
         return node
 
+    def to_json(self):
+        return {
+            'node_id': self.node_id,
+            'label': self.label,
+            'acl': self.acl,
+            'properties': self.properties,
+            'system_annotations': self.system_annotations,
+            'edges_out': [e.to_json() for e in self.edges_out],
+            'edges_in': [e.to_json() for e in self.edges_in]
+        }
+
+    @classmethod
+    def from_json(cls, node_json):
+        Type = Node.get_subclass(node_json['label'])
+        return Type(node_id=node_json['node_id'],
```

Comment:
I'm guessing that you're calling `Node.from_json`, if so, then `cls` will always be `Node`.  Changing `node.label` cannot change the class of `node`.  If you were to call `Case.from_json`, on a doc with `{"label": "case"}`, then you would correctly get a `Case` back, but if `{"label": "sample"}`, you would get the same error.  The reason you can _even_ set `.label` is because psqlgraph v1 didn't have subclasses, just labels, and v2 made an attempt to be as backwards compatible as possible, this is why the function `PolyNode` was created: to be a drop in replacement function for the original node constructor `Node.__init__()`